src/protocol/README.md

```diff
@@ -0,0 +1,42 @@
+## Description of protocol
+
+This is not a complete description, but describes the very basics. Contributions
+are welcome.
+
+There are two types of protocol used. One is older and used for legacy purposes,
+and a newer version.
+
+Both protocols have a header and include at least two fields: type and length.
+All the types and what kind of data it includes are described in
+SFSCommunication.h. The format in the comments are:
+
+- name:\[bits\], where the bits part describes what integer type it is (for
```

Comment:
The lowest memory addressable unit is a byte anyway (on x86 at least), so most probably it's always a multiple of 8 bits (unless you are dealing with obscure/special hardware).

But I agree, it would be confusing right now with the way we have it structured in the header files. We might want to change them for bytes but it would definitely be out of scope for this documentation